### PR TITLE
TINY-10708: Suppress bedrock logs in remote testing

### PR DESCRIPTION
**Pull request by danoaky-tiny**
Related Ticket: TINY-10708

Description of Changes:
- Reduce logs when on CI to 10% intervals.
- Cursor is no longer moved (or at least tries to) when on CI, so logs are just normally printed.

Pre-checks:
* [x] Changelog entry added
* [X] package.json versions have not been changed (done by Lerna on release)
* [X] ~Tests have been added (if applicable)~

Before merging:
* [x] Ensure internal dependencies are on appropriate versions
  * For stable releases, all dependencies must be stable
  * For release candidates, all dependencies must be release candidates or stable

### CODE REVIEW

**Review by TheSpyder — CHANGES_REQUESTED** [review]
Looks pretty good. Rather than using remote as the flag, can you please use a TTY check and maybe look for the CI environment variable?

I want this change for all CI, including headless tests.

File: modules/server/src/main/ts/bedrock/server/Controller.ts [hunk 1]
```diff
@@ -107,9 +108,16 @@ export const create = (stickyFirstSession: boolean, singleTimeout: number, overa
...
+    if (stickyFirstSession && (timeoutError || session.id !== stickyId)) return false;
+    if (!Env.IS_CI || session.done) return true;
+    // Only update the HUD at 10% intervals on remote:
+    return session.results.length % Math.round(session.totalTests * 0.1) === 0;
+  };
+
   const updateHud = (session: TestSession) => {
-    if (!outputToHud) return;
-    if (stickyFirstSession && (timeoutError || session.id !== stickyId)) return;
+    if (!shouldUpdateHud(session)) return;
```
Comment (TheSpyder) [inline modules/server/src/main/ts/bedrock/server/Controller.ts:120]: I'm torn about whether to keep this on the server or just reduce the frequency of traffic from the client.

Lower traffic should speed up the connection on slow tunnels, but might risk incorrect HUD output when a test times out.

**Review by TheSpyder — APPROVED** [review]
Much better 👍

File: modules/server/src/main/ts/bedrock/server/Controller.ts (hunk 1 above)
Comment (danoaky-tiny) [reply modules/server/src/main/ts/bedrock/server/Controller.ts:120]: Maybe it should always log tests that didn't pass as well?
```typescript
if (!Env.IS_CI || session.done || !session.results.at(-1)?.passed) return true;
```

File: modules/server/src/main/ts/bedrock/cli/Hud.ts [hunk 2]
```diff
@@ -25,24 +26,29 @@ export const create = (testfiles: string[], loglevel: 'simple' | 'advanced'): Hu
...
-    readline.clearLine(stream, 1);
+    if (!Env.IS_CI) {
+      readline.clearLine(stream, 1);
+    }
     return Promise.resolve();
   };
 
-  const advUpdate = (data: ResultData) => {
-    if (started) {
+  const advUpdate = async (data: ResultData) => {
```
Comment (ltrouton) [inline modules/server/src/main/ts/bedrock/cli/Hud.ts:41]: I can't see any `await` or promises
```suggestion
  const advUpdate = (data: ResultData) => {
```

**Review by ltrouton — APPROVED** [review]

File: modules/server/src/main/ts/bedrock/server/Controller.ts (hunk 1 above)
Comment (TheSpyder) [reply modules/server/src/main/ts/bedrock/server/Controller.ts:120]: What would be really nice is
* reduce frequency of client to HUD status push to 10% in CI
* when there is a test failure, override this and do a push immediately

Comment (TheSpyder) [reply modules/server/src/main/ts/bedrock/server/Controller.ts:120]: But this is at least _better_, so I think publish it and if you like work on that idea as a new PR.

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
+### Improved
+
+- Reduced progress logs when on CI to 10% intervals #TINY10708
```
Comment (TheSpyder) [inline CHANGELOG.md:11]:
```suggestion
- Reduced progress logs when on CI to 10% intervals #TINY-10708
```

**Pull request merged**